src/Core/Session/Model/UserSession.php

```diff
@@ -140,9 +140,9 @@ public function isSiteAdmin(): bool
 	}
 
 	/** {@inheritDoc} */
-	public function setVisitorsContacts()
+	public function setVisitorsContacts(string $my_url)
```

Comment:
Can you explain why this is necessary? Everytime I see set this string, we set it per session as well, or did I miss a place?

Reply:
As it turns out, in one special case we weren't setting this in the session object but in the `$_SESSION` array, which may be distinct depending on the node configuration. Since this method is always called at the same time the session variable is set with the value, there was no need for a call to `Session->get` and an explicit parameter was simpler and would have avoided this issue in the first place.

In a way it did help replacing the last unwarranted `$_SESSION` occurrences but even then it's giving the original code a lot more credit than it deserves. 😄